components/Header.js

```diff
@@ -1,23 +1,111 @@
 import styled from "styled-components";
+import Link from "next/link";
 
 const NavContainer = styled.header`
   display: flex;
+  position: fixed;
   justify-content: center;
   align-items: center;
+  top: 0px;
   height: 60px;
   background-color: #202020;
   color: #f2f2f2;
   width: 100%;
+  z-index: 9999;
 `;
 
 const HeaderTitle = styled.h1`
   font-size: 2rem;
   margin: 0;
+  z-index: 9888;
 `;
 
-export default function Header() {
+const NavbarMenu = styled.nav`
+  position: fixed;
+  top: 60px;
+  left: 0;
+  width: 250px;
+  height: 100%;
+  background: #f2f2f2;
+  z-index: 8888;
+  transition: transform 0.3s ease-in-out;
+  transform: ${({ isOpen }) =>
+    isOpen ? "translateX(0%)" : "translateX(-100%)"};
+`;
+
+const NavUnderContainer = styled.nav`
+  position: fixed;
+  top: 60px;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: #000;
+  opacity: 0.5;
+  z-index: 7777;
+  transition: transform 0.3s ease-in-out;
+  transform: ${({ isOpen }) =>
+    isOpen ? "translateX(0)" : "translateX(-100%)"};
+`;
+
+const NavbarMenuItem = styled.ul`
```

Comment:
I think the name "NavbarMenuItem" could be misunderstood. For me, it would be a list item and not the list itself, so you might think about renaming it